vms/proposervm/block.go

```diff
@@ -164,13 +164,25 @@ func (p *postForkCommonComponents) Verify(
 		)
 	}
 
-	return p.vm.verifyAndRecordInnerBlk(
+	if err := p.vm.verifyAndRecordInnerBlk(
 		ctx,
 		&smblock.Context{
 			PChainHeight: parentPChainHeight,
 		},
 		child,
-	)
+	); err != nil {
+		return err
+	}
+
+	if err := p.vm.State.PutVerifiedBlock(child.ID()); err != nil {
+		return err
+	}
+
+	if err := p.vm.State.PutBlock(child, choices.Processing); err != nil {
+		return err
+	}
```

Comment:
Hmm maybe I'm not understanding something. I think we still need to put into the verified block index (where we're just storing the block id, not the full block), in addition to storing the full block in the block index because in order to iterate back towards the last accepted block you need to know each block's parent which is only available in the full block data which isn't stored in the verified block index.

Reply:
I thought the full block was stored in the verified block index, but it's just the IDs. Marking as resolved, cus it's better as is.